modules/ocf_hpc/facts.d/nvidia_gpu_info

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+import json
+import re
+import shutil
+import subprocess
+
+
+def main():
+    if not shutil.which('nvidia-smi'):
+        return
+
+    try:
+        nvidia_smi_output = subprocess.run(['nvidia-smi', '--query-gpu=name,index', '--format=csv,noheader'],
+                                           stdout=subprocess.PIPE,
+                                           universal_newlines=True,
+                                           check=True,
+                                           ).stdout
+    except subprocess.CalledProcessError:
+        return
+
+    nvidia_gpu_list = nvidia_smi_output.split('\n')[:-1]
+
+    nvidia_gpu_info_dict = {}
+
+    for gpu_string in nvidia_gpu_list:
+        # Used for determining what the numbers of the GPU's device files in /dev
+        gpu_index = int(gpu_string.split(',')[1])
+        # Strip all except numbers and add 'nv' in front to distinguish from number of GPUs
+        # when specifying type in SLURM.
+        gpu_model_number = 'nv' + re.sub('[^0-9]', '', gpu_string.split(',')[0])
```

Comment:
Instead of calling split twice I'd do this...

```python
gpu_model, gpu_index = gpu_string.split(',', 1)
gpu_model_number = 'nv' + re.sub('[^0-9]', '', gpu_model)
nvidia_gpu_info_dict[int(gpu_index)] = gpu_model_number
...
```

Doing that split with the regex and string concatenation makes the code unnecessarily confusing imo. Honestly wouldn't hurt to put the regex deletion in a helper method called `strip_ints`.